File: Server/src/main/java/core/game/content/dialogue/GemMerchantDialogue.java
package core.game.content.dialogue;

import core.game.node.entity.npc.NPC;
import core.plugin.Initializable;
import core.game.node.entity.player.Player;

/**
 * Represents the gem merchant dialogue plugin.
 * @author 'Vexia
 * @version 1.0
 */
@Initializable
public final class GemMerchantDialogue extends DialoguePlugin {

	/**
	 * Constructs a new {@code GemMerchantDialogue} {@code Object}.
	 */
	public GemMerchantDialogue() {
		/**
		 * empty.
		 */
	}

	/**
	 * Constructs a new {@code GemMerchantDialogue} {@code Object}.
	 * @param player the player.
	 */
	public GemMerchantDialogue(Player player) {
		super(player);
	}

	@Override
	public DialoguePlugin newInstance(Player player) {
		return new GemMerchantDialogue(player);
	}

	@Override
	public boolean open(Object... args) {
		npc = (NPC) args[0];
		interpreter.sendDialogues(npc, FacialExpression.HALF_GUILTY, "Here, look at my lovely gems.");
		stage = 0;
		return true;
	}

	@Override
	public boolean handle(int interfaceId, int buttonId) {
		switch (stage) {
		case 0:
			end();
			npc.openShop(player);
			break;
		}
		return true;
	}

	@Override
	public int[] getIds() {
		return new int[] { 570 };
	}
}

File: Server/src/main/content/region/kandarin/ardougne/dialogue/GemMerchantDialogue.kt
package content.region.kandarin.ardougne.dialogue

import core.api.openNpcShop
import core.game.dialogue.DialoguePlugin
import core.game.dialogue.FacialExpression
import core.game.node.entity.npc.NPC
import core.game.node.entity.player.Player
import core.plugin.Initializable
import core.tools.START_DIALOGUE
import org.rs09.consts.NPCs

@Initializable
class GemMerchantDialogue(player: Player? = null) : DialoguePlugin(player) {

    override fun newInstance(player: Player?): DialoguePlugin {
        return GemMerchantDialogue(player)
    }

    override fun open(vararg args: Any?): Boolean {
        npc = args[0] as NPC
        npcl(
            FacialExpression.HAPPY, "Here, look at my lovely gems."
        ).also { stage = START_DIALOGUE }
        return true
    }

    override fun handle(interfaceId: Int, buttonId: Int): Boolean {
        when (stage) {
            0 -> {
                openNpcShop(player, NPCs.GEM_MERCHANT_570)
                end()
            }
        }
        return true
    }

    override fun getIds(): IntArray {
        return intArrayOf(NPCs.GEM_MERCHANT_570)
    }
}